### ISSUE TITLE
[Bug]: zsh exited due to tagget pointers error when trying to invoke edit-command-line

### ISSUE THREAD

**Issue by marcosfnsc** [termux/termux-packages#22383]
### Problem description

similar to #7332

I run zsh as my default shell, in its settings I defined that when I press ESC the current command is taken to a neovim buffer

the problematic configuration in zshrc:
```shell
export EDITOR=nvim
autoload -U edit-command-line
zle -N edit-command-line
bindkey '\033' edit-command-line # Use ESC to edit the current command line:
export KEYTIMEOUT=1
```
I've never had any problems with this until now, when I press the defined key Termux gives the following error and closes the shell
![vlcsnap-2024-11-24-21h31m34s611](https://github.com/user-attachments/assets/3e8f5a12-975e-406c-958c-ef0bf059937f)

when I change the value of bindkey to '040'  (the space bar) the error also occurs

### What steps will reproduce the bug?

install zsh and edit .zshrc with the lines about edit-command-line and press ESC

### What is the expected behavior?

open a neovim buffer with the command that was in the shell

### System information

```shell
Termux Variables:
TERMUX_API_VERSION=0.50.1
TERMUX_APK_RELEASE=F_DROID
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_APP_PID=22310
TERMUX_IS_DEBUGGABLE_BUILD=0
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.1
TERMUX__USER_ID=0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://repository.su/termux/termux-main/ stable main
# x11-repo (sources.list.d/x11.list)
deb https://repository.su/termux/termux-x11/ x11 main
Updatable packages:
All packages up to date
termux-tools version:
1.44.1
Android version:
14
Kernel build information:
Linux localhost 5.4.259-qgki-g3ad529ef5970 #1 SMP PREEMPT Thu Oct 17 09:55:30 UTC 2024 aarch64 Android
Device manufacturer:
Xiaomi
Device model:
22111317PG
LD Variables:
LD_LIBRARY_PATH=
LD_PRELOAD=/data/data/com.termux/files/usr/lib/libtermux-exec.so
Installed termux plugins:
com.termux.nix versionCode:188036
com.termux.api versionCode:51
```

**Comment by TomJo2000** [comment]
Please check `pkg list zsh` and verify that you have `5.9-7`, we just fixed a regression causing some patches to not get properly applied for 5.9-6 in #22381

**Comment by TomJo2000** [comment]
I can confirm from my testing that this is fixed in `5.9-7`

**Comment by marcosfnsc** [comment]
indeed I am running zsh 5.9-6, thanks
